slack/rtm/client.py

```diff
@@ -448,37 +459,27 @@ async def _dispatch_event(self, event, data=None):
                     # close/error callbacks.
                     break
 
-                if inspect.iscoroutinefunction(callback):
+                if self.run_async or inspect.iscoroutinefunction(callback):
                     await callback(
```

Comment:
This is a great catch. In future major releases, we may be able to clearly say "if you go with `run_sync=True`, all callbacks must be coroutines." but it's not that timing when we release a minor version. Also, we don't need to have this change to resolve the existing concurrency issues.

I will just revert this change.

Reply:
reverted by https://github.com/slackapi/python-slackclient/pull/662/commits/d9aa23802e92fe3f61ff2fc90271af3b941b9e84

Reply:
with this change reverted, when `run_async=False` the callback will still be invoked as a coroutine (using the `await`). that will also lead to problems. it seems like the way to fix this would be to detect when `run_async=True and !inspect.iscoroutinefunction(callback)` to throw an error.

Reply:
Thanks, letting developers notice doing wrong seems a nice addition. I've added some tests and added an error in https://github.com/slackapi/python-slackclient/pull/662/commits/227f9490ab8908c2cddcf07d1500c5cbf8fb5b4d